sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -673,6 +677,245 @@ def test_check_estimator_get_tags_default_keys():
+                                          random_state=0)
+    y_test = y[-test_size:]
+
+    class BaseMultiLabelClassifierMock(ClassifierMixin, BaseEstimator):
+        def __init__(self, response_output):
+            self.response_output = response_output
+
+        def fit(self, X, y):
+            return self
+
+    class MultiLabelClassifierPredict(BaseMultiLabelClassifierMock):
+        def predict(self, X):
+            return self.response_output
+
+    class MultiLabelClassifierPredictProba(BaseMultiLabelClassifierMock):
+        def predict_proba(self, X):
+            return self.response_output
+
+    class MultiLabelClassifierDecisionFunction(BaseMultiLabelClassifierMock):
+        def decision_function(self, X):
+            return self.response_output
+
+    # 1.method predict
+    # 1.1 inconsistent array type
+    clf = MultiLabelClassifierPredict(response_output=y_test.tolist())
+    err_msg = (
+        r"MultiLabelClassifierPredict.predict is expected to output a "
+        r"NumPy array. Got <class 'list'> instead."
+    )
+    assert_raises_regex(
+        AssertionError,
+        err_msg,
+        check_classifiers_multilabel_format_output,
+        clf.__class__.__name__,
+        clf,
+    )
+    # 1.2. inconsistent shape
+    clf = MultiLabelClassifierPredict(response_output=y_test[:, :-1])
+    err_msg = (
+        r"MultiLabelClassifierPredict.predict output a NumPy array of "
+        r"shape \(25, 4\) instead of \(25, 5\)."
+    )
+    assert_raises_regex(
+        AssertionError,
+        err_msg,
+        check_classifiers_multilabel_format_output,
+        clf.__class__.__name__,
+        clf,
+    )
+    # 1.3 inconsistent dtype
+    clf = MultiLabelClassifierPredict(
+        response_output=y_test.astype(np.float64)
+    )
+    err_msg = (
+        r"MultiLabelClassifierPredict.predict does not output the same "
+        r"dtype than the targets."
+    )
+    assert_raises_regex(
+        AssertionError,
+        err_msg,
+        check_classifiers_multilabel_format_output,
+        clf.__class__.__name__,
+        clf,
+    )
+
+    # 2. method predict_proba
+    # 2.1 unknown output type
+    clf = MultiLabelClassifierPredictProba(
+        response_output=sp.csr_matrix(y_test)
+    )
+    err_msg = (
+        r"Unknown returned type <class 'scipy.sparse.csr.csr_matrix'> by "
+        r"MultiLabelClassifierPredictProba.predict_proba. A list or a Numpy "
+        r"array are expected."
+    )
+    assert_raises_regex(
+        ValueError,
+        err_msg,
+        check_classifiers_multilabel_format_output,
+        clf.__class__.__name__,
+        clf,
+    )
+    # 2.2 for list output
+    # 2.2.1 iconsistent length
+    clf = MultiLabelClassifierPredictProba(response_output=y_test.tolist())
+    err_msg = (
+        "When MultiLabelClassifierPredictProba.predict_proba returns a list, "
+        "the list should  be of length n_outputs and contain NumPy array. Got "
+        "length of 25 instead of 5."
```

Comment:
As with [the previous comment on interpolating strings](https://github.com/scikit-learn/scikit-learn/pull/19859/files#r617474175).
```suggestion
        f"length of {test_size} instead of {n_outputs}."
```